src/components/MessageInput/hooks/useTimer.ts

```diff
@@ -0,0 +1,23 @@
+import { useEffect, useState } from 'react';
+
+export const useTimer = ({ startFrom }: { startFrom: number }) => {
+  const [secondsLeft, setSecondsLeft] = useState<number | undefined>();
+
+  useEffect(() => {
+    let countdownTimeout: ReturnType<typeof setTimeout>;
+    if (typeof secondsLeft === 'number' && secondsLeft > 0) {
+      countdownTimeout = setTimeout(() => {
+        setSecondsLeft(secondsLeft - 1);
```

Comment:
You can use the callback variant of setState and remove effect dependency.

Reply:
How would I then decide to prevent setting timeout, when the value is 0? I understand that the author wanted to avoid setInterval due to throttling. Recursion with setTimeout may be a problem.

Reply:
Ah, I see, I haven't noticed that `secondsLeft` is used outside of the state update. We can keep it like this then :)